### ISSUE TITLE
Old version on FlatHub

### ISSUE THREAD

**Issue by SokoloffA** [flathub/com.github.Flacon#7]
I haven't found a way to write to you privately, so I'm writing here.
First of all, many thanks you for building the package for my program!

Your repository is listed as the publisher one on [flacon flathub page](https://flathub.org/apps/details/com.github.Flacon). And I see that you are building new versions very fast, but on the flathub page contains the very old version of the program (5.4.0 was released 27 Aug 2019). 

Could you tell me what the version update policy is on the flathub site?

**Comment by Eonfge** [comment]
I'm working on fixing that. Guess there is something wrong with the listing, because this morning I distributed Flacon 6.0.0

@barthalion You know if there is something wrong with the store page?

**Comment by Eonfge** [comment]
Before I forget, you can track the updates here:
https://github.com/flathub/com.github.Flacon/commits/master

I try to stick to close to the latest version, so I'm never more then a few days behind on the official repository.

**Comment by SokoloffA** [comment]
> I try to stick to close to the latest version, so I'm never more then a few days behind on the official repository.

You're awesome! I get notifications, and I see you do it faster than the light!
I wanted to write about flatpak on the [download page](https://flacon.github.io/download/), and I noticed an oddity on the flatpak page.

**Comment by bilelmoussaoui** [comment]
@SokoloffA Let us know if you would like to have write access to this repository :)

**Comment by Eonfge** [comment]
@bilelmoussaoui You any idea about the version number on the Flathub page? I pushed the update an hour ago but it doesn't seem to change

**Comment by Eonfge** [comment]
Found it! The cause of the issue lies deep in the belly of the code.

@SokoloffA I see that there is actually an xml in the project:
https://github.com/flacon/flacon/blob/master/misc/com.github.Flacon.metainfo.xml.in

I had forgotten this, and I caused a caching conflict by adding my own com.github.Flacon.metadata.xml

I fixed the issue for the time by renaming my matadata.xml to metainfo.xml, causing an override of your version. How this happened? 

https://github.com/flacon/flacon/issues/102

So, long story short, shall I make a pullrequest to fix the metainfo for the next release?

**Comment by SokoloffA** [comment]
I am not very familiar with the flatpak and especially with the requirements of ftalhub. 
Maybe it's better to remove this file from my repository?

**Comment by barthalion** [comment]
Metainfo/appdata files essentially belong to the project code, they aren't Flathub specific. This way you can add releases info at the time of actual release.

**Comment by Eonfge** [comment]
@SokoloffA see the pull request containing the latest live version:
https://github.com/flacon/flacon/pull/121

**Comment by SokoloffA** [comment]
I merged your request. But, I know myself, I will forget to update this file before release. And after the release, I can't update this file, because the old version is already in the release. 

I suggest automating this. I see two ways:
1. Automatically generate metainfo.xml when building a flatpak package. We can use the [github API](https://api.github.com/repos/flacon/flacon/releases) for this purpose. We can cahange the run scrip in your repo.
2. I will add a git hook on to update the file in my repository. 

The first option is more interesting, because you can use it in other packages that you support.

What do you think?

**Comment by Eonfge** [comment]
Not sure what would work best. I can try to help you remember, if you want? For Flacon 6, you first published a beta which I didn't distribute. If you do so as well next time, I can send you a pull request for an updated metainfo

**Comment by SokoloffA** [comment]
I'll try to make both, then we'll decide.

**Comment by Eonfge** [comment]
Forgot to close this one. If somebody still has some further comments, feel free to make a new issue.
